scripts/perf/perf_kit/sqlalchemy.py

```diff
@@ -15,31 +15,59 @@
 # specific language governing permissions and limitations
 # under the License.
 
-import contextlib
 import os
 import time
 import traceback
+from typing import Callable
 
 from sqlalchemy import event
 
 
-@contextlib.contextmanager
-def trace_queries(display_time=True, display_trace=True, display_sql=False, display_parameters=True):
+def _pretty_format_sql(text: str):
+    import pygments
+
+    from pygments.formatters.terminal import TerminalFormatter
+    from pygments.lexers.sql import SqlLexer
+    text = pygments.highlight(
+        code=text, formatter=TerminalFormatter(), lexer=SqlLexer()
+    ).rstrip()
+    return text
+
+
+class TraceQueries:
     """
-    Tracking SQL queries in a code block. The result is displayed directly on the screen - ``print``
+    Tracking SQL queries in a code block.
 
+    :param display_no: If True, displays the query number.
     :param display_time: If True, displays the query execution time.
     :param display_trace: If True, displays the simplified (one-line) stack trace
     :param display_sql: If True, displays the SQL statements
     :param display_parameters: If True, display SQL statement parameters
-    :return:
+    :param print_fn: The function used to display the text. By default,``builtins.print``
     """
-    import airflow.settings
-
-    def before_cursor_execute(conn, cursor, statement, parameters, context, executemany):
+    def __init__(
+        self,
+        *,
```

Comment:
what's this for?

Reply:
This forces keyword arguments to be used.  https://www.python.org/dev/peps/pep-3102/

Reply:
Nice, did not know that

Reply:
It's well visualized here.
![Screenshot 2020-05-07 at 13 53 29](https://user-images.githubusercontent.com/12058428/81291527-32b6df00-906a-11ea-8220-460535a296e3.png)
https://deepsource.io/blog/python-positional-only-arguments/

Reply:
Yeah, very useful. Py3 only of course.